src/goto-symex/symex_target_equation.cpp

```diff
@@ -368,7 +368,9 @@ unsigned int symex_target_equationt::clear_assertions()
 
   for (SSA_stepst::iterator it = SSA_steps.begin(); it != SSA_steps.end(); it++)
   {
-    if (it->type == goto_trace_stept::ASSERT)
+    bool user_provided =
```

Comment:
This is confusing. You claim that if `smt_during_symex` is enabled, the user will never provide the assertion.

I think the idea here is to avoid erasing the assertion when doing incremental SMT, right? If so, why don't you simply use the flag  `smt_during_symex`  in the if-condition?

Reply:
When doing incremental SMT, the assertion is indeed needed for lemma learning.

Reply:
Sorry, the variable name is misleading. My question is whether certain assertions, like internally generated checks (e.g., bounds checks), need to be removed during incremental BMC?

Reply:
No, they don't need to be removed. These assertions can help the SMT solver when performing incremental learning.